packages/client/src/stripHeaders.ts

```diff
@@ -0,0 +1,36 @@
+import { HTTPHeaders } from './links/core';
+
+// from https://github.com/nodejs/undici/blob/9e5316c8b04a7b35522d0d5b8de71f67fa2a3c34/lib/fetch/constants.js
+const forbiddenHeaderNames = [
+  'accept-charset',
+  'accept-encoding',
+  'access-control-request-headers',
+  'access-control-request-method',
+  'connection',
+  'content-length',
+  'cookie',
+  'cookie2',
+  'date',
+  'dnt',
+  'expect',
+  'host',
+  'keep-alive',
+  'origin',
+  'referer',
+  'te',
+  'trailer',
+  'transfer-encoding',
+  'upgrade',
+  'via',
+];
+
+/**
+ * Strips an HTTP header object of internally reserved keys (for undici/Node.js built-in fetch since v18)
+ * @param headers Your HTTP header object
+ * @returns Stripped headers without blacklisted keys
+ */
+export const stripHeaders = (headers: HTTPHeaders): HTTPHeaders =>
```

Comment:
I don't like this name. Maybe `stripInvalidHeaders`?

Reply:
I wouldn't go with `undici` because they're just following the Fetch API spec. [MDN calls them "forbidden"](https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_header_name) so I'm leaning towards `stripForbiddenHeaders`

Reply:
> Forbidden header names start with Proxy- or Sec-, or are one of the following names:

Cool, let's use that list instead of unidici then as well and make sure we strip headers starting with `proxy-` or `sec-` too?

Reply:
@KATT Good point, I'd stick with the array from undici (unless it's missing something) and add a check for the forbidden string prefixes